.github/workflows/release.yaml

```diff
@@ -0,0 +1,54 @@
+# This is a GitHub workflow defining a set of jobs with a set of steps.
+# ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
+#
+name: Release
+
+# Always tests wheel building, but only publish to PyPI on pushed tags.
+on:
+  pull_request:
+    paths-ignore:
+      - "**.md"
+      - ".github/workflows/*.yaml"
+      - "!.github/workflows/release.yaml"
+  push:
+    paths-ignore:
+      - "**.md"
+      - ".github/workflows/*.yaml"
+      - "!.github/workflows/release.yaml"
+    branches-ignore:
+      - "dependabot/**"
+      - "pre-commit-ci-update-config"
+    tags: ["**"]
+  workflow_dispatch:
+
+jobs:
+  build-release:
+    runs-on: ubuntu-24.04
+    permissions:
+      # id-token=write is required for pypa/gh-action-pypi-publish, and the PyPI
+      # project needs to be configured to trust this workflow.
+      #
+      # ref: https://github.com/jupyterhub/team-compass/issues/648
+      #
+      id-token: write
```

Comment:
@ryanlovett @yuvipanda @minrk could one of you provide this github workflow with the permissions to publish to PyPI?

You'd visit https://pypi.org/manage/project/jupyter-rsession-proxy/settings/publishing/ and under "Add a new publisher" populate 3/4 fields like:
- owner: jupyterhub
- repo: jupyter-rsession-proxy
- workflow name: release.yaml
- environment: leave unspecified